cpp/src/io/text/multibyte_split.cu

```diff
@@ -383,17 +383,31 @@ std::unique_ptr<cudf::column> multibyte_split(cudf::io::text::data_chunk_source
     stream,
     streams);
 
+  // String offsets point to the first character of a field
+  // This finds the first field whose first character starts inside or after the byte range
```

Comment:
This is a great comment. It makes it very clear to me where the bug was introduced: 
> ... or after the byte range